test/prow/env.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// env.go provides a central point to read all environment variables defined by Prow.
+
+package prow
+
+import (
+	"fmt"
+
+	"github.com/kelseyhightower/envconfig"
+)
+
+type EnvConfig struct {
+	CI          bool   `required:"true"`
+	Artifacts   string `required:"true"`
+	JobName     string `required:"true" split_words:"true"`
+	JobType     string `required:"true" split_words:"true"`
+	JobSpec     string `required:"true" split_words:"true"`
+	BuildID     string `required:"true" envconfig:"BUILD_ID"`
+	ProwJobID   string `required:"true" envconfig:"PROW_JOB_ID"`
+	RepoOwner   string `split_words:"true"`
+	RepoName    string `split_words:"true"`
+	PullBaseRef string `split_words:"true"`
+	PullBaseSha string `split_words:"true"`
+	PullRefs    string `split_words:"true"`
+	PullNumber  uint   `split_words:"true"`
+	PullPullSha string `split_words:"true"`
+}
+
+func GetEnvConfig() (*EnvConfig, error) {
+	var ec EnvConfig
+	if err := envconfig.Process("", &ec); err != nil {
+		return nil, fmt.Errorf("failed getting environment variables for Prow: %w", err)
```

Comment:
Is this s a runtime error?

Reply:
I think an error makes more sense as people can also call this function from a non-CI environment. In that case, it's expected to have an error and I think an error is better than a `panic`.

Reply:
So what will happen in CI? Is it going to fail all CI jobs if this ever failed? What could make it throw?

Reply:
Not sure if I understand your question correctly. 
There are two cases:

1. If this function is called from a non-CI environment, it will always return an error, I have changed the function to return an error for such scenario in a different way.

2. If the function is called from a CI environment and it returns an error, it means Prow has some breaking change in the required fields, see https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables, which should be rare, but in this case, we'll need to make the corresponding update here.

Reply:
Sorry for not making it clear. I assume this function will be called in all Prow jobs in the future, and don't think it's generally the best idea to take Prow down because of possibly irrelevant API breakage. MY personally preference would be a unit test to capture this rather than failing at runtime. Can we make this function to do best effort or have a fall back strategy so it doesn't fail that hard?

Reply:
I got what you meant and I agree with it.
The easiest change is to remove all `required:"true"` from the struct tags so there won't be an error simply because of Prow spec changes. Please let me know if you have any further thoughts.

Reply:
sgtm. Please make a unit test so that it can capture broken API

Reply:
Hmm, what do you expect the unit test to do? Checking all required env vars are not empty? IMO such unit test is not useful in preventing breaking changes, and will block other PRs if one day Prow indeed makes one of the required env vars as optional

Reply:
What I was thinking about is your previous implementation, if possible we could use that one as a unit test. But I agree with you that we should not block pkg PRs because of this reason

Reply:
Hmm, so what's your current suggestion now that I changed the previous implementation?

Reply:
I think it's good for now